### PR TITLE
Fix/uniform agent advance

### PR DESCRIPTION
**Pull request by andre-merzky**
This is a more thorough fix for #3162 which ensures that failed and cancelled tasks are always pushed back to the client side.

### CODE REVIEW

**Review by mtitov — COMMENTED** [review]
couple comments inline and one here

- should the below code have this value "tmgr_pending"? (or even being removed)

https://github.com/radical-cybertools/radical.pilot/blob/79ecf1d165f876df61bfc6890ffd68d3810d3925/src/radical/pilot/tmgr/staging_input/default.py#L262

File: src/radical/pilot/agent/scheduler/base.py [hunk 1]
```diff
@@ -411,9 +411,9 @@ def control_cb(self, topic, msg):
                         self._raptor_tasks[queue].remove(task)
 
             for task in to_cancel:
-                task['target_state'] = rps.CANCELED
                 task['control']      = 'tmgr_pending'
                 task['$all']         = True
+
```
[the remark this answers is not in the document]
Comment (andre-merzky) [reply src/radical/pilot/agent/scheduler/base.py:416]: Yeah, removed it.

**Comment by andre-merzky** [comment]
Thanks @mtitov , I cleaned this up a bit more.

**Comment by andre-merzky** [comment]
> couple comments inline and one here
> 
>     * should the below code have this value "tmgr_pending"? (or even being removed)
> 
> 
> https://github.com/radical-cybertools/radical.pilot/blob/79ecf1d165f876df61bfc6890ffd68d3810d3925/src/radical/pilot/tmgr/staging_input/default.py#L262

Discussed and agree to leave as is.

**Pull request merged**